node/src/p2p_msg_handler.rs

```diff
@@ -0,0 +1,306 @@
+                Ok(_) => {}
+                Err(e) => {
+                    println!("finish_subscribe_topic: send request NodeInfo {e}");
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::action::{GOATMessage, GOATMessageContent, NodeInfo, send_to_peer};
+    use crate::env::get_rpc_support_actors;
+    use crate::middleware::AllBehaviours;
+    use crate::middleware::swarm::{
+        Bitvm2Swarm, Bitvm2SwarmConfig, MessageHandler, TickMessageType,
+    };
+    use crate::utils::save_node_info;
+    use bitvm2_lib::actors::Actor;
+    use libp2p::gossipsub::MessageId;
+    use libp2p::{PeerId, Swarm};
+    use prometheus_client::registry::Registry;
+    use store::localdb::LocalDB;
+    use tokio_util::sync::CancellationToken;
+    use tracing::log::info;
+    use tracing::warn;
+
+    struct MockBitvmSwarmMessageHandler {
+        pub local_db: LocalDB,
+    }
+    pub async fn detect_heart_beat(
+        swarm: &mut Swarm<AllBehaviours>,
+        node_info: NodeInfo,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        tracing::info!("start detect_heart_beat");
+        let message_content = GOATMessageContent::RequestNodeInfo(node_info);
+        // send to actor
+        let actors = get_rpc_support_actors();
+        for actor in actors {
+            match send_to_peer(swarm, GOATMessage::from_typed(actor, &message_content)?) {
+                Ok(_) => {}
+                Err(err) => warn!("{err}"),
+            }
+        }
+        Ok(())
+    }
+    impl MessageHandler for MockBitvmSwarmMessageHandler {
+        async fn recv_and_dispatch(
+            &self,
+            _swarm: &mut Swarm<AllBehaviours>,
+            actor: Actor,
+            from_peer_id: PeerId,
+            id: MessageId,
+            message: &[u8],
+        ) -> anyhow::Result<()> {
+            tracing::info!("recv_and_dispatch");
+            if id == GOATMessage::default_message_id() {
+                tracing::info!("recv_and_dispatch receive local message");
+                return Ok(());
+            }
+            let message: GOATMessage = serde_json::from_slice(message)?;
+            let content: GOATMessageContent = message.to_typed()?;
+            info!(
+                "actor:{actor}, peer_id:{from_peer_id}, content:{}",
+                String::from_utf8_lossy(&message.content)
+            );
+
+            if let (GOATMessageContent::RequestNodeInfo(node_info), _) = (content, actor) {
+                save_node_info(&self.local_db, &node_info).await.expect("save_node_info");
+            }
+            Ok(())
+        }
+
+        async fn handle_tick_message(
+            &self,
+            swarm: &mut Swarm<AllBehaviours>,
+            _peer_id: PeerId,
+            actor: Actor,
+            msg_type: TickMessageType,
+        ) -> anyhow::Result<()> {
+            tracing::info!("handle_tick_message :{}", msg_type.to_string());
```

Comment:
use #[tracing::instrument(level = Level::INFO)] to decorate function handle_tick_message, instead of printing the `handle_tick_message `.